yb-voyager/cmd/common.go

```diff
@@ -308,6 +318,73 @@ func retrieveMigrationUUID(exportDir string) error {
 	return nil
 }
 
+func getMetaDBPath(exportDir string) string {
+	return filepath.Join(exportDir, "metainfo", "meta.db")
+}
+
+func createAndInitMetaDBIfRequired(exportDir string) error {
+	metaDBPath := getMetaDBPath(exportDir)
+	if utils.FileOrFolderExists(metaDBPath) {
+		// already created and initied.
+		return nil
+	}
+	err := createMetaDBFile(metaDBPath)
+	if err != nil {
+		return err
+	}
+	err = initMetaDB(metaDBPath)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func createMetaDBFile(path string) error {
+	f, err := os.Create(path)
+	if err != nil {
+		return fmt.Errorf("not able to create meta db file :%w", err)
+	}
+	err = f.Close()
+	if err != nil {
+		return fmt.Errorf("error while closing meta db file: %w", err)
+	}
+	return nil
+}
+
+func initMetaDB(path string) error {
+	conn, err := sql.Open("sqlite3", path)
+	if err != nil {
+		return fmt.Errorf("error while opening meta db :%w", err)
+	}
+	cmds := []string{
+		fmt.Sprintf(`CREATE TABLE %s (segment_no INTEGER, size_committed INTEGER);`, QUEUE_SEGMENT_META_TABLE_NAME),
```

Comment:
Do we need `IF NOT EXISTS` clause?

Reply:
Add a column named `file_path`. It should be segment file path relative to export-dir.

Reply:
We only run the DDLs when we init the database for the first time, so `IF NOT EXISTS` is not needed! 

Added `file_path` but how would it help 🤔 queue segment no should be enough, no?